### PR TITLE
chore(scripts): Add `--dist` option to `clean.sh` and minor cleanup in `scripts/*`

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
Title says it all

I was working on some scripts and debugging locally. Did some minor changes and formatting. Putting here to decrease the scope of #14283 

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **Chores**
	- Improved script formatting by ensuring a newline at the end of certain scripts.
	- Enhanced flexibility and functionality in CI scripts by introducing variables and conditional checks.
	- Added a new option to the clean script for more granular control over file cleaning, including distribution files.

### CODE REVIEW

**Comment by AndesKrrrrrrrrrrr** [comment]
@coderabbitai review

**Comment by AndesKrrrrrrrrrrr** [comment]
https://github.com/coderabbitai review

**Comment by AndesKrrrrrrrrrrr** [comment]
@coderabbitai review

File: scripts/clean.sh [hunk 1]
```diff
@@ -83,6 +93,12 @@ clean_caches() {
   dry || rm -rf "${CLEAN_CACHES_LIST[@]}"
 }
 
+clean_dist() {
+
+  dry || rm -rf -- **/dist/
+
+}
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply scripts/clean.sh]: @coderabbitai review

[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply scripts/clean.sh]: Fixed in new commit

File: scripts/ci/40_e2e.sh [hunk 2]
```diff
@@ -1,4 +1,12 @@
 #!/bin/bash
 set -euxo pipefail
 
-yarn nx run "$1":e2e-ci
+APP="${1:-$APP}"
```
Comment (AndesKrrrrrrrrrrr) [inline scripts/ci/40_e2e.sh:4]: Nonsensical self-reference :angry: 
```suggestion
APP="${1:-$APP}"
```

File: scripts/ci/90_docker-express.sh [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
 set -euxo pipefail
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
 # shellcheck disable=SC1091
 source "$DIR"/_common.sh
 
 # Building Docker images for ExpressJS-based apps
+if command -v podman &>/dev/null; then
```
Comment (AndesKrrrrrrrrrrr) [inline scripts/ci/90_docker-express.sh:10]: What if you have both `docker` and `podman`?
```suggestion
if { ! command -v docker &>/dev/null } && { command -v podman &>/dev/null }; then
```

Comment (AndesKrrrrrrrrrrr) [reply scripts/ci/90_docker-express.sh:10]: The `_docker.sh` script should maybe be modified to allow either. And Podman should implement the missing options (or at least not fail)

File: scripts/ci/40_e2e.sh (hunk 2 above)
Comment (AndesKrrrrrrrrrrr) [reply scripts/ci/40_e2e.sh:4]: No, this makes sense. Fails if neither `APP` is sett nor an argument provided.

File: scripts/clean.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply scripts/clean.sh]: Please re-review, since I've updated since you requested changes.

**Review by robertaandersen — APPROVED** [review]
LGTM

**Pull request merged**